app/src/main/java/xyz/jhughes/laundry/fragments/MachineFragment.java

```diff
@@ -164,10 +163,15 @@ public void onResponse(Call<ArrayList<Machine>> call, Response<ArrayList<Machine
                 @Override
                 public void onFailure(Call<ArrayList<Machine>> call, Throwable t) {
                     //likely a timeout -- network is available due to prev. check
-                    showErrorDialog(getString(R.string.error_server_message));
-                    mSwipeRefreshLayout.setRefreshing(false);
-                    isRefreshing = false;
-                    alertNetworkError();
+                    if (isAdded() && getActivity() != null) {
```

Comment:
First you would want a call.isCanceled() (https://square.github.io/retrofit/2.x/retrofit/retrofit2/Call.html#isCanceled--). If its canceled do nothing. Then check the rest. 

Calling cancel on the Call still ends up in onFailure.